zerver/lib/markdown/__init__.py

```diff
@@ -1833,13 +1833,25 @@ def handleMatch(  # type: ignore[override] # supertype incompatible with supersu
         return None, None, None
 
 
-class UserGroupMentionPattern(markdown.inlinepatterns.Pattern):
-    def handleMatch(self, m: Match[str]) -> Optional[Element]:
-        match = m.group(2)
-
+class UserGroupMentionPattern(markdown.inlinepatterns.InlineProcessor):
+    def handleMatch(  # type: ignore[override] # supertype incompatible with supersupertype
+        self, m: Match[str], data: str
+    ) -> Union[Tuple[None, None, None], Tuple[Element, int, int]]:
+        match = m.group(0)
+        match_span = m.span()
         db_data = self.md.zulip_db_data
+
+        # As patterns like @*group name* are captured.
+        start_syntax = "@*"
+        end_syntax = "*"
         if self.md.zulip_message and db_data is not None:
-            name = extract_user_group(match)
+            # As InlineProcessor API sends all the matching part
+            # coming after (?<![^\s\'\"\(,:<]), so @ is also
+            # included in match so it needs to be strip.
+            if match.startswith(start_syntax) and match.endswith(end_syntax):
+                name = match[len(start_syntax) : -len(end_syntax)]
+            else:
+                name = match
```

Comment:
Is this `else` clause possible?  I don't understand under what circumstance it would be, given the regular expression.

(If it is, do we have a test for it?).